src/utils/verbose.h

```diff
@@ -67,7 +67,28 @@ class Env {
         }
     }
 
-    static int getVerbose() { return verbose_value(); }
+    static int getVerbose() { return verboseValue(); }
+
+// Pipeline Parallel
+private:
+    static int &pipelineValue() {
+        static int value = 1;
+        return value;
+    }
+
+public:
+    static void initPipeline() {
+        char *xft_pipeline_value = getenv("XFT_PIPELINE_STAGES");
```

Comment:
Check if MPI_rank is divisible by ppStages?

Reply:
Check in common_decoder.h